tests/device-mqtt/main_test.go

```diff
@@ -1,36 +1,57 @@
 package test
 
 import (
+	"edgex-snap-testing/env"
+	"edgex-snap-testing/utils"
 	"log"
 	"os"
 	"testing"
-
-	"MQTT-test-project/utils"
 )
 
 func TestMain(m *testing.M) {
 
 	log.Println("[GLOBAL SETUP]")
 
-	// TODO
-	// utils.InstallSnaps("edgex-device-mqtt", "edgexfoundry")
-	// utils.RemoveSnaps("edgex-device-mqtt", "edgexfoundry")
-	// the function can be variadic and take zero or more inputs
-	// e.g. https://github.com/canonical/edgex-snap-hooks/blob/50df6237c8eb5b49d497d3a6f978f83391905308/utils.go#L254
-
-	utils.RunCommand(nil,
-		"sudo snap remove --purge edgex-device-mqtt",
-		"sudo snap remove --purge edgexfoundry",
-		"sudo snap install edgexfoundry --channel=latest/stable",
-		"sudo snap install edgex-device-mqtt --channel=latest/stable")
+	// start clean
+	utils.SnapRemove(nil,
+		"edgex-device-mqtt",
+		"edgexfoundry",
+	)
+
+	// install the device snap before edgexfoundry
+	// to catch build error sooner and stop
+	if env.SnapcraftProjectDir == "" {
+		utils.SnapInstall(nil, "edgex-device-mqtt")
+	} else {
+		utils.SnapBuild(nil, env.SnapcraftProjectDir)
+		utils.SnapInstallLocal(nil, env.SnapcraftProjectDir)
+	}
+
+	utils.SnapInstall(nil, "edgexfoundry")
+
+	// for local build, the interface isn't auto-connected.
+	// connect manually regardless
```

Comment:
The auto connection would fail if the interface names are changed. The connect hooks that are triggered as part of the connection may also fail for different reasons. The tests would have to cover those.